src/Symfony/Component/Routing/Router.php

```diff
@@ -210,6 +217,16 @@ public function getContext()
     }
 
     /**
+     * Sets the ConfigCache factory to use.
+     *
+     * @param ConfigCacheFactoryInterface $configCacheFactory The factory to use.
+     */
+    public function setConfigCacheFactory(ConfigCacheFactoryInterface $configCacheFactory)
```

Comment:
Probably a matter of style and I'd be glad to follow your rules.

However, caching is completely optional in the `Router` and so I chose to make this change as unobtrusive as possible.